docs/design/proposals/multi-version/multi-version.md

```diff
@@ -0,0 +1,259 @@
+  (and a mutating webhook handler) even if they are not modified?
+- How do we deprecate/remove versions in a programatical way?
+
+#### CRDs Mono Version vs Group Version
+
+The answer to the questions above will impact the design of [`metadata.yaml`]
+[iam-metadata.yaml]. And how we think about ACK CRD versioning and how
+version deprecation happens. Do we deprecate a set of CRD versions? or only
+one specific CRD version? unfortunetaly picking one of those solutions is kinda
+a hard one way door solution to say the least.
+
+CRD Group versioning:
+
+- Pros:
+1. Current ACK code-generator works this way
+2. Easier code generation (import paths, packaging etc...)
+3. Easier CRD Version difference calculation
+4. Doesn't enormously impact the rest of the features (references etc...)
+5. No need to change `metadata.yaml`
+
+- Cons:
+1. Confusing for users
+2. Little CRD Changes means we will bump all the CRDs versions
+3. Not very kubernetes style.
+
+CRD Mono versioning:
+
+- Pros:
+1. Similar to k8s style
+2. Easier deployments (risk reduced when upgrading CRDs)
+3. Less error zone as we will target resources one by one
+
+- Cons:
+1. Very tricky in code generation
+2. Needs a rethinking of generator.yaml + metadata.yaml
+3. We don't know the impact on existing features (references etc...)
+
+#### CRDs version deprecation
+
+Curently we only have `metadata.yaml` as a tool to deprecate versions, see
+APIInfoStatuses contants in [code-generator/pkg/metadata][code-gen/pkg/md]. Which
+works very well with `pkg/model/multiversion` package. In case we want to go with
+CRD Mono Version way, we will have to change this model to be Resource specific
+rather than version specific.
+
+### Webhook server
+
+Controller runtime already offers webhook server setup as a builtin mechanisms, and
+it was ported to ACK runtime/code-generator earlier in [this PR][ack-webhook-pr].
+The missing part here is a certificate issuer like Cert Manager - behind the scenes
+the webhook server needs to authenticate the controllers in order to allow them to
+mutate/validate CRs.
+The Cert Manager part will be discuss bellow.
+
+### Code Generation
+
+https://github.com/aws-controllers-k8s/code-generator/pull/122
+[TODO AMINE EXPAND THIS SECTION]
+
+### Cert manager
+
+Cert-manager plays a critical role in securing and authenticating communication between
+different components of the Cluster. It is essential to provide valid certificates for the
+webhook server to ensure secure communication between the webhook server and the Controllers
+API server. Without valid certificates, the webhook server cannot be authenticated, which
+can result in a failure to register the webhooks. Therefore, it is imperative to use 
+cert-manager to ensure that the webhook server has a valid certificate before registering
+any webhooks.
+
+Cert manager will probably be added as an optional install to the helm charts, and will
```

Comment:
Why "probably" here?  Is this question really open for debate?  Is there any feasible alternative that we should discuss?